tsl/src/chunk_api.c

```diff
@@ -1370,45 +1376,71 @@ fetch_remote_chunk_stats(Hypertable *ht, FunctionCallInfo fcinfo, bool col_stats
 				 errmsg("function returning record called in context "
 						"that cannot accept type record")));
 
+	sqlcmd = deparse_func_call(fcinfo);
+	rs = async_request_set_create();
 	data_nodes = ts_hypertable_get_data_node_name_list(ht);
-	cmdres = ts_dist_cmd_invoke_func_call_on_data_nodes(fcinfo, data_nodes);
 
-	/* Expect TEXT response format since dist command API currently defaults
-	 * to requesting TEXT */
+	foreach (lc, data_nodes)
+	{
+		const char *node_name = lfirst(lc);
+		AsyncRequest *req;
+		TSConnection *connection =
+			data_node_get_connection(node_name, REMOTE_TXN_NO_PREP_STMT, true);
+		req = async_request_send(connection, sqlcmd);
+		/* Set single-row mode in order to reduce memory usage in case result
+		 * set is big */
+		async_request_set_single_row_mode(req);
+		async_request_attach_user_data(req, (char *) node_name);
+		async_request_set_add(rs, req);
+	}
+
 	tf = tuplefactory_create_for_tupdesc(tupdesc, true);
-	num_rows = ts_dist_cmd_total_row_count(cmdres);
-	/* Estimate the number of non-duplicate stats to use for initial size of
+
+	/* Assume 500 non-duplicate stats to use for initial size of
 	 * StatsProcessContext. Use slightly bigger than strictly necessary to
 	 * avoid a resize. */
-	num_stats = (5 * num_rows) / (ht->fd.replication_factor * 4);
+	stats_process_context_init(&statsctx, 500);
+	old_mcxt = MemoryContextSwitchTo(statsctx.per_tuple_mcxt);
 
-	stats_process_context_init(&statsctx, num_stats);
-
-	for (i = 0; /* exit when res == NULL below */; i++)
+	while (true)
```

Comment:
Some compilers will treat a `while (true)` as a commented-out block. While this hasn't an issue for more modern compilers, the generally-accepted convention for infinite loops (or loops that have flow control internally) is to use `for (;;)` instead, which additionally eliminates static condition check warnings the compiler emits.